packages/appstore/src/components/modals/real-wallets-upgrade/real-wallets-upgrade.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { DesktopWrapper, MobileDialog, MobileWrapper, Modal, Button } from '@deriv/components';
+import { useWalletMigration } from '@deriv/hooks';
+import { ContentFlag } from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { observer, useStore } from '@deriv/stores';
+import { WalletsIntro } from './components/wallets-intro/wallets-intro';
+import ReadyToUpgradeWallets from './components/ready-to-upgrade-wallets';
+import './real-wallets-upgrade.scss';
+import WalletLinkingStep from './wallet-linking-step/wallet-linking-step';
+import getMockWalletMigrationResponse from '../../../constants/mock_wallet_migration_response';
+
+const RealWalletsUpgrade = observer(() => {
+    const { traders_hub } = useStore();
+    const { is_real_wallets_upgrade_on, toggleWalletsUpgrade, content_flag } = traders_hub;
+    const is_eu = content_flag === ContentFlag.EU_REAL || content_flag === ContentFlag.EU_DEMO;
+
+    const [current_step, setCurrentStep] = React.useState(0);
+    const [is_disabled, setIsDisabled] = React.useState(false);
+
+    React.useEffect(() => {
+        if (!is_real_wallets_upgrade_on) {
+            setCurrentStep(0);
+            setIsDisabled(false);
+        }
+    }, [is_real_wallets_upgrade_on]);
+
+    const handleNext = () => setCurrentStep(prev_step => prev_step + 1);
+
+    const handleBack = () => setCurrentStep(prev_step => prev_step - 1);
+
+    const handleClose = () => toggleWalletsUpgrade(false);
+
+    const { start_migration } = useWalletMigration();
+
+    const upgradeToWallets = () => {
+        start_migration();
+        toggleWalletsUpgrade(false);
+    };
+
+    const toggleCheckbox = React.useCallback(() => {
+        setIsDisabled(prevDisabled => !prevDisabled);
+    }, []);
+
+    const DefaultFooter = React.useMemo(
+        () => (
+            <Modal.Footer className='wallet-steps__footer' has_separator>
+                <Button secondary large className='wallet-steps__footer-button' onClick={handleBack}>
+                    {localize('Back')}
+                </Button>
+                <Button primary large className='wallet-steps__footer-button' onClick={handleNext}>
+                    {localize('Next')}
+                </Button>
+            </Modal.Footer>
+        ),
+        []
+    );
+
+    const InitialFooter = React.useMemo(
+        () => (
+            <Modal.Footer className='wallet-steps__footer' has_separator>
+                <Button secondary large className='wallet-steps__footer-button' onClick={handleClose}>
+                    {localize('Maybe later')}
+                </Button>
+                <Button primary large className='wallet-steps__footer-button' onClick={handleNext}>
+                    {localize('Next')}
+                </Button>
+            </Modal.Footer>
+        ),
+        []
+    );
```

Comment:
Defeats the purpose of useMemo without dependencies no?

Reply:
No, it just mean it will not be changed from the 1st render of the component. Anyway, it was refactored